src/JSInterop/Microsoft.JSInterop/src/JSRuntimeBase.cs

```diff
@@ -58,6 +65,22 @@ public Task<T> InvokeAsync<T>(string identifier, params object[] args)
             }
         }
 
+        private async Task<T> GetResultTask<T>(long taskId, string methodIdentifier, Task<T> task)
+        {
+            await Task.WhenAny(task, Task.Delay(DefaultAsyncCallTimeout.Value));
+
+            if (!task.IsCompleted)
+            {
+                _pendingTasks.TryRemove(taskId, out _);
+                throw new JSException($"The call to {methodIdentifier} exceded the maximum allowed wait time.");
```

Comment:
https://github.com/davidfowl/AspNetCoreDiagnosticScenarios/blob/master/AsyncGuidance.md#using-a-timeout

This should also throw `OperationCanceledException`

Reply:
Don't you think a more specific exception might make troubleshooting easier (specially when seeing exceptions on deployed applications)

Reply:
https://docs.microsoft.com/en-us/dotnet/api/system.timeoutexception?view=netframework-4.8

Reply:
Either way it shouldn't be `JSException` - that's the one we use to represent an exception thrown by js code.